vue-components/src/components/LookupMenu.vue

```diff
@@ -9,10 +9,12 @@
 			:key="index"
 			v-for="(menuItem, index) in menuItems"
 			:class="{
-				'wikit-LookupMenu__item--selected': index === selectedItemIndex
+				'wikit-LookupMenu__item--selected': index === selectedItemIndex,
+				'wikit-LookupMenu__item--active': index === activeItemIndex,
 			}"
 			@click="$emit( 'select', menuItem )"
-			@mousedown.prevent
+			@mousedown.prevent="activeItemIndex = index"
+			@mouseup="activeItemIndex = -1"
```

Comment:
It looks good. However, I think we should avoid adding more javascript to a problem that is caused by javascript. I think the addition of this `preventDafault` on mousedown might have been an oversight in the past,and if we need to prevent default browser behaviour, it's probably in that `onBlur` method we created yesterday to overcome the problem (this would also move this focus responsibility back to the parent, where it belongs)

Reply:
I very much agree with the general sentiment of not doing with JavaScript what can be done with CSS.

However, I don't see how _that_ functionality belongs into the parent. Quite the opposite, it breaks the logic-encapsulation of the child, as every parent component that uses LookupMenu is now forced to add a handler for `onBlur` because otherwise the onSelect will never fire for mouse clicks as the LookupMenu will already have disappeared. That seems rather like a good example of leaking internal logic.

Otherwise, it would be the responsibility of that `onBlur`-handler to figure out whether the focus-loss has happened due to an mouse-down above an item in the LookupMenu or not in order determine whether the LookupMenu should still be shown. That seems a strange thing for a parent component to be worrying about.

Reply:
While still having different views on some other aspects, @itamargiv and I agreed that the LookupMenu will unavoidably require a number of extra logic in its parent component anyway, so adding to that list isn't too much of an antipattern anymore.
This extra logic that is required for the use of LookupMenu will need good documentation. It will probably be best to add that extra documentation in the course of extracting the Dropdown out of the Lookup.